### PR TITLE
[Tests] Bumps `chromedriver` to v107

### PR DESCRIPTION
**Pull request by tmarkley**
### Description
* Aligns all versions of `chromedriver`
* Removes manual resolution of `axios`
 
### Issues Resolved
N/A - follow-up from https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1410 and related to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1298
 
### Check List
- [x] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by tmarkley** [comment]
Dependency tree for `axios`, this is how I addressed this manual resolution:

```
$ yarn why axios
yarn why v1.22.15
[1/4] Why do we have the module "axios"...?
[2/4] Initialising dependency graph...
warning Resolution field "typescript@4.0.2" is incompatible with requested version "typescript@~4.5.2"
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "axios@0.27.2"
info Has been hoisted to "axios"
info Reasons this module exists
   - "workspace-aggregator-59198173-6b89-4e89-98f7-bc432bb9c629" depends on it
   - Hoisted from "_project_#@osd#ui-shared-deps#axios"
   - Hoisted from "_project_#@osd#dev-utils#axios"
info Disk size without dependencies: "616KB"
info Disk size with unique dependencies: "728KB"
info Disk size with transitive dependencies: "1.09MB"
info Number of shared dependencies: 6
=> Found "chromedriver#axios@0.24.0"
info This module exists because "_project_#chromedriver" depends on it.
info Disk size without dependencies: "540KB"
info Disk size with unique dependencies: "588KB"
info Disk size with transitive dependencies: "588KB"
info Number of shared dependencies: 1
Done in 1.35s.

$ npm list axios
opensearch-dashboards@3.0.0 /workspaces/OpenSearch-Dashboards
├─┬ @osd/dev-utils@1.0.0 -> /workspaces/OpenSearch-Dashboards/packages/osd-dev-utils
│ └── axios@0.27.2 
├─┬ @osd/ui-shared-deps@1.0.0 -> /workspaces/OpenSearch-Dashboards/packages/osd-ui-shared-deps
│ └── axios@0.27.2  deduped
└─┬ chromedriver@100.0.0
  └── axios@0.24.0 
```

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -152,9 +152,9 @@ jobs:
           npm uninstall -g yarn
           npm i -g yarn@1.22.10
 
-      # image has the latest chrome v99
+      # match version in package.json
       - name: Setup chromedriver
-        run: yarn add --dev chromedriver@99.0.0
+        run: yarn add --dev chromedriver@107.0.3
```
Comment (tmarkley) [inline .github/workflows/build_and_test_workflow.yml:157]: @kavilla how do we bump this version in the Docker image?

**Comment by tmarkley** [comment]
I didn't event see Miki's PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2990. I'll remove the changes to the workflow config.

File: .github/workflows/build_and_test_workflow.yml (hunk 1 above)
Comment (AMoo-Miki) [reply .github/workflows/build_and_test_workflow.yml]: @tmarkley I have #2990 to automate this at runtime.

Comment (tmarkley) [reply .github/workflows/build_and_test_workflow.yml]: @AMoo-Miki I just noticed that PR this morning, it's great! I've already updated this PR to remove these parts.

**Review by AMoo-Miki — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Comment by joshuarrrr** [comment]
It seems like we should have backported this to active branches - let me know if that's wrong.

**Comment by tmarkley** [comment]
> It seems like we should have backported this to active branches - let me know if that's wrong.

@joshuarrrr yeah I would've added those labels but I don't have permissions.

**Pull request merged**